src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -2,145 +2,143 @@ package cromwell.engine
+                                 binding: WdlBinding,
+                                 actualInputs: WorkflowCoercedInputs,
+                                 backend: Backend) extends Actor {
 
-  private var maybeSymbolStore: Option[SymbolStore] = None
-  private var maybeExecutionStore: Option[ExecutionStore] = None
-  private var maybeBackend: Option[Backend] = None
   private val log = Logging(context.system, this)
 
-  private def notStarted: Receive = {
-    LoggingReceive {
-      case Start(backend) =>
-        maybeExecutionStore = Option(new ExecutionStore(binding))
-        maybeSymbolStore = Option(new SymbolStore(binding, actualInputs))
-        maybeBackend = Option(backend)
-        // Poll for runnable calls.
-        context.system.scheduler.schedule(
-          initialDelay = 0 milliseconds,
-          interval = 5 seconds,
-          receiver = self,
-          message = CheckExecutionStatus)(context.system.dispatcher)
-
-        context.become(started)
-        context.parent ! Started
-
-      case bad@_ =>
-        sender ! InvalidOperation(s"Received message $bad before any Start message")
-    }
-  }
+  private val storeActor = context.actorOf(StoreActor.props(binding, actualInputs))
 
-  private def createRequiredInputs(call: Call): Unit = {
-    call.inputMappings.foreach { case(inputName, expression) =>
-      val ast = expression.ast.asInstanceOf[Ast]
-      val Seq(lhs, rhs) = Seq("lhs", "rhs").map { ast.getAttribute(_).asInstanceOf[Terminal].getSourceString }
-      val outputFqn = Seq(call.parent.get.name, lhs, rhs).mkString(".")
-      val inputFqn = Seq(call.parent.get.name, call.name, inputName).mkString(".")
-      symbolStore.copyOutputToInput(outputFqn, inputFqn)
-    }
+  def receive: Receive = unstarted
+
+  /** Unstarted `Receive` state that only responds to `Start` messages. */
+  def unstarted: Receive = LoggingReceive {
+    case WorkflowActor.Start => handleInitialStart()
+    case bad => fail(bad, "unstarted")
   }
 
-  private def started: Receive = {
-    LoggingReceive {
-
-      // TODO: I'd prefer that receive partial functions are expressable in one (perhaps 2) lines, even if that means pulling stuff into a func
-
-      case CheckExecutionStatus =>
-        if (executionStore.isDone) {
-          context.parent ! Done(symbolStore)
-        } else {
-          val runnableCalls = executionStore.runnableCalls
-          if (runnableCalls.nonEmpty) {
-            val namesOfCallsBeingStarted = runnableCalls.map { _.name}.toSeq.sorted
-            log.info("Starting calls: " + namesOfCallsBeingStarted.mkString(", "))
-            runnableCalls.foreach { call =>
-              createRequiredInputs(call)
-              startCallActor(symbolStore, call)
-            }
-          }
-        }
+  /** Started `Receive` state. */
+  def started: Receive = LoggingReceive {
+    case CallActor.Started(call) =>
+      storeActor ! UpdateStatus(call, ExecutionStatus.Running)
```

Comment:
This is interesting... I think I might be missing something here.  So you get a message from the CallActor saying that it started.  Then you update the execution status in the symbol store... I'm confused about why this wouldn't go in the execution store directly, especially considering that's what the SymbolStore seems to do:

```
case UpdateStatus(call, status) =>
      executionStore.updateStatus(call, status)
```